presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -659,7 +659,7 @@ public boolean testDouble(double value)
                     return false;
                 }
             }
-            return true;
+            return !Double.isNaN(value) && !Double.isNaN(lower) && !Double.isNaN(upper);
```

Comment:
@bhhari My reading of this change suggests that if either `lower` or `upper` is NaN, then the filter is effectively always-false. Is this so? That doesn't seem right. If lowerUnbounded = true, then the value of lower shouldn't matter. Similarly, if upperUnbounded = true, then the value of upper shouldn't matter. If lowerUnbounded = false and lower is NaN, then the filter is effectively always-false. Similarly, upperUnbounded = false and upper is NaN makes the filter always-false. In these cases, it makes more sense to use ALWAYS_FALSE than an instance of DoubleRangeFilter. 

Based on the above, I suggest the following changes:

- update DoubleRange's constructor to check lower and upper values for NaN

```
            checkArgument(lowerUnbounded || !Double.isNaN(lower));
            checkArgument(upperUnbounded || !Double.isNaN(upper));
```

- update TupleDomainFilterUtils#doubleRangeToFilter to return ALWAYS_FALSE

```
        if (!low.isLowerUnbounded() && Double.isNaN(lowerDouble)) {
            return ALWAYS_FALSE;
        }

        if (!high.isUpperUnbounded() && Double.isNaN(upperDouble)) {
            return ALWAYS_FALSE;
        }
```

- update testDouble to check for NaN in the beginning of the method

```
        @Override
        public boolean testDouble(double value)
        {
            if (Double.isNaN(value)) {
                return false;
            }
```

What do you think?

Reply:
@mbasmanova  the ALWAYS_FALSE approach is much better